lib/polyfill/patchedmediakeys_20140218.js

```diff
@@ -0,0 +1,800 @@
+ *
+ * @constructor
+ * @param {string} keySystem
+ * @implements {MediaKeys}
+ */
+shaka.polyfill.PatchedMediaKeys.v20140218.MediaKeys = function(keySystem) {
+  shaka.log.debug('v20140218.MediaKeys');
+
+  this.keySystem_ = keySystem;
+  this.nativeMediaKeys_ = new MSMediaKeys(keySystem);
+
+  /** @private {!shaka.util.EventManager} */
+  this.eventManager_ = new shaka.util.EventManager();
+};
+
+
+/** @override */
+shaka.polyfill.PatchedMediaKeys.v20140218.MediaKeys.prototype.
+    createSession = function() {
+  shaka.log.debug('v20140218.MediaKeys.createSession');
+
+  // Alias
+  var v20140218 = shaka.polyfill.PatchedMediaKeys.v20140218;
+
+  return new v20140218.MediaKeySession(this.nativeMediaKeys_);
+};
+
+
+/** @override */
+shaka.polyfill.PatchedMediaKeys.v20140218.MediaKeys.prototype.
+    setServerCertificate = function(serverCertificate) {
+  shaka.log.debug('v20140218.MediaKeys.setServerCertificate');
+
+  // There is no equivalent in v20140218, so return failure.
+  return Promise.reject(new Error('setServerCertificate not supported on ' +
+      'this platform.'));
+};
+
+
+
+/**
+ * An implementation of MediaKeySession.
+ *
+ * @constructor
+ * @param {MSMediaKeys} nativeMediaKeys_
+ * @implements {MediaKeySession}
+ * @extends {shaka.util.FakeEventTarget}
+ */
+shaka.polyfill.PatchedMediaKeys.v20140218.
+    MediaKeySession = function(nativeMediaKeys_) {
+  shaka.log.debug('v20140218.MediaKeySession');
+  shaka.util.FakeEventTarget.call(this, null);
+
+  // Native MediaKeySession, which will be created in generateRequest
+  this.nativeMediaKeySession_ = null;
+
+  this.nativeMediaKeys_ = nativeMediaKeys_;
+
+  // Promises that are resolved later
+  this.generateRequestPromise_ = null;
+  this.updatePromise_ = null;
+
+  /** @type {string} */
+  this.sessionId = '';
+
+  /** @type {number} */
+  this.expiration = NaN;
+
+  /** @type {!shaka.util.PublicPromise} */
+  this.closed = new shaka.util.PublicPromise();
+
+  /** @type {!MediaKeyStatusMap} */
+  this.keyStatuses =
+      new shaka.polyfill.PatchedMediaKeys.v20140218.MediaKeyStatusMap();
+};
+goog.inherits(shaka.polyfill.PatchedMediaKeys.v20140218.MediaKeySession,
+    shaka.util.FakeEventTarget);
+
+
+/** @override */
+shaka.polyfill.PatchedMediaKeys.v20140218.MediaKeySession.prototype.
+    generateRequest = function(initDataType, initData) {
+  shaka.log.debug('v20140218.MediaKeySession.generateRequest');
+
+  this.generateRequestPromise_ = new shaka.util.PublicPromise();
+
+  try {
+    // This EME spec version requires a MIME content type as the 1st param
+    // to createSession, but doesn't seem to matter what the value is...
+    this.nativeMediaKeySession_ = this.nativeMediaKeys_
+        .createSession('video/mp4', initData);
+
+    // Alias
+    var v20140218 = shaka.polyfill.PatchedMediaKeys.v20140218;
+
+    // Attach session event handlers here
+    this.nativeMediaKeySession_.addEventListener('mskeymessage',
+        v20140218.onMsKeyMessage_.bind(this));
```

Comment:
If you bind them to this, why aren't these handlers private methods on the prototype for MediaKeySession?